Add a warning when using source = "."

hduelme opened the pull request:
I implemented #203. As suggested for each source parameter a quick fix to replace the source parameter is given.

thunderhook commented on src/main/java/org/mapstruct/intellij/inspection/ThisUsedAsSourcePropertyInspection.java:
```diff
@@ -0,0 +1,105 @@
+            return;
+        }
+        List<LocalQuickFix>  fixes = new ArrayList<>();
+        PsiMethod annotatedMethode = getAnnotatedMethod( psiAnnotation );
```
nitpick: `annotatedMethode->annotatedMethod`

hduelme replied:
Fixed

thunderhook commented on testData/inspection/ThisUsedAsSourcePropertyInspection.java:
```diff
@@ -0,0 +1,52 @@
+    @Mapping(target = "source", source = <warning descr="''.'' should not be used as a source.">"."</warning>)
+    Target map(Source source);
+}
+
```
Could you please add another test with a method like this:
```java
    Target map(Source source, Long age, @Context String thisMustNotBeSuggested);
```

To test that there are multiple quick fixes and that context variables do not get suggested?
I tried it "by hand" and it works, but a test for regression would do no harm. 👍

hduelme replied:
Test added

thunderhook reviewed (commented):
Thanks @hduelme! Works great!

Just a nit-pick and a note about an additional test. Thanks in advance!

hduelme commented:
@thunderhook thanks for the review. I also added tests if there is no source property and for the case `*` is not used as a source. In addition I added the missing documentation for the new inspection

thunderhook commented:
Very nice, thanks for implementing it so quickly @hduelme .
